apps/skilavottord/ws/src/app/modules/samgongustofa/models/samgongustofa.service.ts

```diff
@@ -0,0 +1,336 @@
+                  // otherwise is 'inUse'
+                  if (car['vehiclestatus'][0] == 'Afskráð') {
+                    carStatus = 'deregistered'
+                    carIsRecyclable = false
+                  }
+                  let carHasCoOwner = true
+                  if (car['otherowners'][0] == '0') {
+                    carHasCoOwner = false
+                  }
+
+                  // Create new Vehicle information object on each vehicle
+                  const vehicleObj = new VehicleInformation(
+                    car['permno'][0],
+                    car['type'][0],
+                    car['color'][0],
+                    car['firstregdate'][0],
+                    carIsRecyclable,
+                    carHasCoOwner,
+                    carStatus,
+                  )
+                  vehicleArr.push(vehicleObj)
+                },
+              )
+              return vehicleArr
+            })
+            .catch(function (err) {
+              loggerReplacement.error(err)
+              return []
+            })
+        })
+        .catch(function (err) {
+          loggerReplacement.error(err)
+          return []
+        })
+
+      this.logger.info('Finished extracting all vehicles')
+      // TODO: connect to database and get 'pending' status
+      const newVehicleArr = this.vehicleInformationList
+      let isRequestSuccess = true
+
+      // ForEach vehicle in vehicleInformationList, check and update vehicle's status
+      for (let i = 0; i < newVehicleArr.length; i++) {
+        const carObj = newVehicleArr[i]
+        this.logger.info(
+          `Starting extracting details information on ${carObj['permno']}`,
+        )
+        if (carObj['status'] == 'inUse') {
+          // Vehicle information
+          // Vehicle information's Soap body
+          const xmlBasicInfoBodyStr = `<soapenv:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:usx="https://xml.samgongustofa.is/scripts/WebObjects.dll/XML.woa/1/ws/.USXMLWS">
+            <soapenv:Header/>
+            <soapenv:Body>
+              <usx:basicVehicleInformation soapenv:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
+                  <username xsi:type="xsd:string">${soapUsername}</username>
+                  <password xsi:type="xsd:string">${soapPassword}</password>
+                  <xmlVersion xsi:type="xsd:string">8</xmlVersion>
+                  <clientPersidno xsi:type="xsd:string">${nationalId}</clientPersidno>
+                  <permno xsi:type="xsd:string">${carObj['permno']}</permno>
+                  <regno xsi:type="xsd:string">null</regno>
+                  <vin xsi:type="xsd:string">null</vin>
+              </usx:basicVehicleInformation>
+            </soapenv:Body>
+          </soapenv:Envelope>`
+
+          const httpService1 = new HttpService()
```

Comment:
This is injected, should not be re-initialised.

Reply:
Done